packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-radio-group/gux-form-field-radio-group.tsx

```diff
@@ -0,0 +1,139 @@
+ */
+@Component({
+  styleUrl: 'gux-form-field-radio-group.scss',
+  tag: 'gux-form-field-radio-group-beta',
+  shadow: true
+})
+export class GuxFormFieldRadioGroupBeta {
+  private disabledObserver: MutationObserver;
+  private label: HTMLLabelElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   *  radio group has error text.
+   */
+  @State()
+  private hasGroupError: boolean = false;
+
+  /**
+   *  radio group has help text.
+   */
+  @State()
+  private hasGroupHelp: boolean = false;
+
+  /**
+   * Disables the radio buttons in the group.
+   */
+  @Prop()
+  disabled: boolean = false;
+
+  @Watch('hasGroupError')
+  watchGroupError(hasGroupError: boolean): void {
+    const radioSlots = this.root.querySelectorAll('gux-form-field-radio');
+    if (radioSlots) {
+      radioSlots.forEach(item => {
+        item.hasGroupError = hasGroupError;
+      });
+    }
+  }
+
+  @Watch('disabled')
+  watchDisabled(): void {
+    this.setDisabledRadio();
+  }
+
+  @OnMutation({ childList: true, subtree: true })
+  onMutation(): void {
+    this.hasGroupError = hasSlot(this.root, 'group-error');
+    this.hasGroupHelp = hasSlot(this.root, 'group-help');
+  }
+
+  componentWillLoad(): void {
+    this.hasGroupError = hasSlot(this.root, 'group-error');
+    this.hasGroupHelp = hasSlot(this.root, 'group-help');
+    this.setLabel();
+    this.setDisabledRadio();
+
+    trackComponent(this.root);
+  }
+
+  disconnectedCallback(): void {
+    if (this.disabledObserver) {
+      this.disabledObserver.disconnect();
+    }
+  }
+
+  private setDisabledRadio(): void {
+    const radioSlots = this.root.querySelectorAll('gux-form-field-radio');
+    if (radioSlots) {
+      radioSlots.forEach(item => {
+        item.hasGroupDisabled = this.disabled;
+      });
+    }
+  }
+
+  private renderScreenReaderText(
+    text: string,
+    condition: boolean = false
+  ): JSX.Element {
+    if (condition) {
+      return (
+        <gux-screen-reader-beta>{text}</gux-screen-reader-beta>
+      ) as JSX.Element;
+    }
+  }
+  render(): JSX.Element {
+    return (
+      <GuxFormFieldFieldsetContainer labelPosition="above">
+        <GuxFormFieldLegendLabel
+          position="above"
+          required={false}
+          labelText={this.label?.textContent}
+        >
+          <slot name="group-label" onSlotchange={() => this.setLabel()} />
+
+          {this.renderScreenReaderText(
+            getSlotTextContent(this.root, 'group-error'),
+            this.hasGroupError
+          )}
+          {this.renderScreenReaderText(
+            getSlotTextContent(this.root, 'group-help'),
+            this.hasGroupHelp
+          )}
+        </GuxFormFieldLegendLabel>
+        <slot />
+        <GuxFormFieldError show={this.hasGroupError}>
+          <slot name="group-error" />
+        </GuxFormFieldError>
+        <GuxFormFieldHelp show={!this.hasGroupError && this.hasGroupHelp}>
+          <slot name="group-help" />
```

Comment:
For some reason `group-help` doesn't make me think of something text related, but maybe I'm overthinking it. I'm inclined to suggest naming this to `group-help-text` and other references of `group-help` being `group-help-text`. Thoughts?

Reply:
I think `group-help` is fine, especially as it is used as a pair with `group-error`